Tests/UnitTests/CompatibilityTests.cpp

```diff
@@ -0,0 +1,666 @@
+#include <Babylon/AppRuntime.h>
+#include <Babylon/ScriptLoader.h>
+#include <Babylon/Polyfills/AbortController.h>
+#include <Babylon/Polyfills/Console.h>
+#include <Babylon/Polyfills/Scheduling.h>
+#include <Babylon/Polyfills/URL.h>
+#include <Babylon/Polyfills/Blob.h>
+#include <gtest/gtest.h>
+#include <future>
+#include <chrono>
+#include <tuple>
+#include <vector>
+#include <string>
+#include <utility>
+#include <atomic>
+
+namespace
+{
+    class EngineCompatTest : public ::testing::Test
+    {
+    protected:
+        Babylon::AppRuntime& Runtime()
+        {
+            if (!m_runtime)
+            {
+                m_runtime = std::make_unique<Babylon::AppRuntime>();
+                m_runtime->Dispatch([](Napi::Env env) {
+                    Babylon::Polyfills::Console::Initialize(env, [](const char*, Babylon::Polyfills::Console::LogLevel) {});
+                    Babylon::Polyfills::AbortController::Initialize(env);
+                    Babylon::Polyfills::Scheduling::Initialize(env);
+                    Babylon::Polyfills::URL::Initialize(env);
+                    Babylon::Polyfills::Blob::Initialize(env);
+                });
+
+                m_loader = std::make_unique<Babylon::ScriptLoader>(*m_runtime);
+            }
+
+            return *m_runtime;
+        }
+
+        Babylon::ScriptLoader& Loader()
+        {
+            if (!m_loader)
+            {
+                Runtime();
+            }
+            return *m_loader;
+        }
+
+        void TearDown() override
+        {
+            m_loader.reset();
+            m_runtime.reset();
+        }
+
+        template<typename T>
+        T Await(std::future<T>& future, std::chrono::milliseconds timeout = std::chrono::milliseconds{5000})
+        {
+            const auto status = future.wait_for(timeout);
+            EXPECT_EQ(status, std::future_status::ready) << "JavaScript did not report back to native code.";
+            if (status != std::future_status::ready)
+            {
+                throw std::runtime_error{"Timeout waiting for JavaScript result"};
+            }
+            return future.get();
+        }
+
+        void Eval(const std::string& script)
+        {
+            Loader().Eval(script.c_str(), "engine-compat");
+        }
+
+    private:
+        std::unique_ptr<Babylon::AppRuntime> m_runtime{};
+        std::unique_ptr<Babylon::ScriptLoader> m_loader{};
+    };
+}
+
+TEST_F(EngineCompatTest, LargeStringRoundtrip)
```

Comment:
We discussed offline about these tests already, but I still don't understand the purpose.

Let's say one of these tests fails. What do we do? Unless the Node-API implementation against a JavaScript VM has an issue, we can't do anything about the problem in this repo. If the Node-API implementation is at fault, we should be adding these tests to the Node-API test suite (i.e., the ones you are bringing over from [this PR](https://github.com/BabylonJS/JsRuntimeHost/pull/116)). If it's a problem with the JavaScript VM itself, it probably should be tested with the JavaScript VM source.

Another way to ask this question. What JavaScript tests make sense to put in this repo? If these go in, what is the criteria for whether more similar tests should be in this repo or not?